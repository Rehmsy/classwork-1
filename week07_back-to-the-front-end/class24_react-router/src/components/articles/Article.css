:local(.article) {
  list-style: none;

  p.image {
    position: relative;
    height: 200px;
    text-align: center;
    
    img {
      height: 100%;
    }
  }
}