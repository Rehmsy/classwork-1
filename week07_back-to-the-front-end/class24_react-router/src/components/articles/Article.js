import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

export default class Article extends Component {

  static propTypes = {
    article: PropTypes.object.isRequired
  };

  render() {
    const { article } = this.props;
    const { title, author, description } = article;
    const { publishedAt, url, urlToImage } = article;

    return (
      <li className={styles.article}>
        
        <h2>{title}</h2>
        <p>by {author} on {new Date(publishedAt).toLocaleString()}</p>
        <p>
          <a href={url} 
            target="_blank" 
            rel="noopener noreferrer">
            view article
          </a>
        </p>

        <p>{description}</p>

        {urlToImage && 
          <p className="image">
            <img src={urlToImage}/>
          </p>
        }
      </li>
    );
  }
}