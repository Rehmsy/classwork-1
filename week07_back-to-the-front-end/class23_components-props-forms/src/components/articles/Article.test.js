import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';
import toJSON from 'enzyme-to-json';

const article = {
  'source': {
    'id': null,
    'name': 'Bitcoinist.com'
  },
  'author': 'Wilma Woo',
  'title': '‘It’ll Happen Anyway’: Andreas Antonopoulos Criticizes Bitcoin ETFs',
  'description': 'Exchange-traded funds (ETFs) “fundamentally undermine the underlying principle of peer-to-peer money,” Andreas Antonopoulos claimed, adopting a critical stance on the financial instrument. ‘Not Your Keys – Not Your Bitcoin’ As part of his monthly Q&A session …',
  'url': 'https://bitcoinist.com/andreas-antonopoulos-criticizes-bitcoin-etfs/',
  'urlToImage': 'https://bitcoinist.com/wp-content/uploads/2018/03/wc-antonopoulos.jpg',
  'publishedAt': '2018-08-15T21:00:04Z'
};

describe('Article', () => {


  it('renders as designed', () => {
    const wrapper = shallow(<Article article={article}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders without image', () => {
    const noImageArticle = { ...article, urlToImage: null };
    const wrapper = shallow(<Article article={noImageArticle}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});

