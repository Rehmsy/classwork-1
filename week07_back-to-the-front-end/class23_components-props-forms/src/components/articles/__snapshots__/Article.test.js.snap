// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders as designed 1`] = `
<li>
  <h2>
    ‘It’ll Happen Anyway’: Andreas Antonopoulos Criticizes Bitcoin ETFs
  </h2>
  <p>
    by 
    Wilma Woo
     on 
    8/15/2018, 2:00:04 PM
  </p>
  <p>
    <a
      href="https://bitcoinist.com/andreas-antonopoulos-criticizes-bitcoin-etfs/"
      rel="noopener noreferrer"
      target="_blank"
    >
      view article
    </a>
  </p>
  <p>
    Exchange-traded funds (ETFs) “fundamentally undermine the underlying principle of peer-to-peer money,” Andreas Antonopoulos claimed, adopting a critical stance on the financial instrument. ‘Not Your Keys – Not Your Bitcoin’ As part of his monthly Q&A session …
  </p>
  <p
    className="image"
  >
    <img
      src="https://bitcoinist.com/wp-content/uploads/2018/03/wc-antonopoulos.jpg"
    />
  </p>
</li>
`;

exports[`Article renders without image 1`] = `
<li>
  <h2>
    ‘It’ll Happen Anyway’: Andreas Antonopoulos Criticizes Bitcoin ETFs
  </h2>
  <p>
    by 
    Wilma Woo
     on 
    8/15/2018, 2:00:04 PM
  </p>
  <p>
    <a
      href="https://bitcoinist.com/andreas-antonopoulos-criticizes-bitcoin-etfs/"
      rel="noopener noreferrer"
      target="_blank"
    >
      view article
    </a>
  </p>
  <p>
    Exchange-traded funds (ETFs) “fundamentally undermine the underlying principle of peer-to-peer money,” Andreas Antonopoulos claimed, adopting a critical stance on the financial instrument. ‘Not Your Keys – Not Your Bitcoin’ As part of his monthly Q&A session …
  </p>
</li>
`;
